### DISCUSSION TITLE
What's the most direct migration path for JedisSentinelPool to begin leveraging UnifiedJedis?

### DISCUSSION THREAD

**Discussion by timmixell** [xetorthio/jedis#2754] [Q&A] [answered]
I'm very excited about the `UnifiedJedis` interface. It looks like there are comparable constructors to facilitate a simple swap of `JedisPool` and `JedisPooled`.  In the absence of a `JedisSentinelPooled` option, I'm having some difficulty locating a constructor in `UnifiedJedis` or it's derivatives that would allow me to easily swap in support for a sentinel environment. It looks like I can pass along a `Jedis.getConnection()` as a constructor arg to `UnifiedJedis`, but I'm wondering if I'm missing something on the sentinel-front.

Thanks in advance,
Tim

**Comment by sazzad16** [comment]
@timmixell You're not missing anything in this regard. There is no easily swap option for sentinel to UnifiedJedis.

**Reply by timmixell** [reply]
Thank you @sazzad16 !

Are there any performance implications/concerns going the route of furnishing a `Jedis.getConnection` from `JedisSentinelPool` as a constructor arg to `UnifiedJedis`? Is there a better route to leverage `UnifiedJedis` in a sentinel environment?

**Reply by sazzad16** [reply]
Implications/concerns:

1. All the implications/concerns of Jedis, e.g. not thread-safe.
2. Operating on invalid connection when the sentinel master is changed.

Better route:

- Copy JedisSentinelPool source and convert it to a ConnectionProvider.

**Comment by shannonantony** [comment]
@timmixell , Hope you are doing well. did you get a chance to convert JedisSentinelPool to ConnectionProvider? If so, do you mind sharing that code? 
thanks

Jay

**Reply by timmixell** [reply]
I haven't had a chance to revisit, unfortunately.

**Comment by sazzad16** [comment, marked as answer]
JedisSentineled is available since Jedis 4.4.0-m1

@timmixell @shannonantony
